lib/modulesync/git.rb

```diff
@@ -39,7 +41,20 @@ def self.update(name, files, message, branch)
         end
       end
       begin
-        repo.commit(message)
+        opts = {}
+        case git_provider
+        when 'gerrit'
+          branch = "HEAD:refs/publish/master/#{branch}"
+          if ! File.exists? ("#{PROJ_ROOT}/#{name}/.git/hooks/commit-msg")
+            Dir.chdir("#{PROJ_ROOT}/#{name}") {
+              %x[gitdir=$(git rev-parse --git-dir); scp -p -P #{git_provider_port} #{git_user}@#{git_provider_address}:hooks/commit-msg ${gitdir}/hooks/]
+            }
+          else
+            opts = {:amend => true}
+            message = nil
```

Comment:
These two lines are extremely problematic and unnecessary. They cause the script to amend the latest commit rather than making a new commit, when the previous commit is likely to be a change already in master and not owned by the committer.